apis/filesystem/src/test/java/org/jclouds/filesystem/integration/FilesystemBlobIntegrationTest.java

```diff
@@ -113,19 +121,59 @@ public void testSetBlobAccess() throws Exception {
    @Override
    protected void checkMPUParts(Blob blob, List<MultipartPart> partsList) {
       assertThat(blob.getMetadata().getETag()).endsWith(String.format("-%d\"", partsList.size()));
-      StringBuilder eTags = new StringBuilder();
+      Hasher eTagHasher = Hashing.md5().newHasher();
       for (MultipartPart part : partsList) {
-         eTags.append(part.partETag());
+         eTagHasher.putBytes(new BigInteger(part.partETag(), 16).toByteArray());
       }
       String expectedETag = new StringBuilder("\"")
-         .append(Hashing.md5().hashString(eTags.toString(), US_ASCII))
+         .append(eTagHasher.hash())
          .append("-")
          .append(partsList.size())
          .append("\"")
          .toString();
       assertThat(blob.getMetadata().getETag()).isEqualTo(expectedETag);
    }
 
+   @Test(groups = { "integration", "live" })
+   public void testMultipartUploadMultiplePartsKnownETag() throws Exception {
+      BlobStore blobStore = view.getBlobStore();
+      String container = getContainerName();
+      // Pre-computed ETag returned by AWS S3 for the MPU consisting of two 5MB parts filled with 'b'
+      String expectedETag = "\"84462a16f6a60478d50148808aa609c1-2\"";
```

Comment:
Just double-checking that the value that S3 returns is also the value that we'd want here, even with the filesystem provider. Assuming it's worth it, have we been able to verify that other providers produce the same value?

Reply:
Ah, just seeing #1251. A follow-up question to that (also /cc @gaul): is S3-compatibility considered a filesystem provider goal? How (if at all) do other blobstore providers treat this case?

Reply:
Good question. Some providers return a similar value, like Swift for static large objects. Others have a different ETag, like Azure, which uses an opaque identifier. The initial proposal I had was to use AWS S3-style ETags because it allows for an easy way to emulate S3 using jclouds (like with the S3Proxy project). That's the motivation behind this PR and the prior one related to it: https://github.com/jclouds/jclouds/pull/1251.

Reply:
> The initial proposal I had was to use AWS S3-style ETags because it allows for an easy way to emulate S3 using jcloud

Clear, thanks!

Reply:
S3 has emerged as the de facto standard for object storage.  I think the average user would prefer the local blobstore acts more similarly to S3, especially since one of the main consumers of this is S3Proxy.  :-)